cylc/flow/task_job_mgr.py

```diff
@@ -235,6 +235,8 @@ def prep_submit_task_jobs(self, workflow, itasks, check_syntax=True):
                 # state transition message reflects the correct submit_num
                 itask.submit_num += 1
                 itask.state_reset(TASK_STATUS_PREPARING)
+                # Avoid clock-expiring an active task:
+                itask.expire_time = None
```

Comment:
I haven't handled clock-expiry of active tasks (I'm implementing the [opt outputs proposal](https://cylc.github.io/cylc-admin/proposal-optional-output-extension.html#proposal)).

Presumably if a task (submit-)fails and has [submission retries configured, then it should be allowed to expire between retries?

Reply:
Yes, I think so.

Reply:
Created an issue for the clock-expire of active tasks thing: https://github.com/cylc/cylc-flow/issues/6025